pkg/controller/statusmanager/status_manager.go

```diff
@@ -195,6 +195,10 @@ func (status *StatusManager) SetFromPods() {
 			progressing = append(progressing, fmt.Sprintf("DaemonSet %q is not available (awaiting %d nodes)", dsName.String(), ds.Status.NumberUnavailable))
 		} else if ds.Status.NumberAvailable == 0 {
 			progressing = append(progressing, fmt.Sprintf("DaemonSet %q is not yet scheduled on any nodes", dsName.String()))
+		} else if ds.Status.UpdatedNumberScheduled < ds.Status.DesiredNumberScheduled {
+			progressing = append(progressing, fmt.Sprintf("DaemonSet %q update is rolling out (%d out of %d updated)", dsName.String(), ds.Status.UpdatedNumberScheduled, ds.Status.DesiredNumberScheduled))
+		} else if ds.Generation > ds.Status.ObservedGeneration {
```

Comment:
The DaemonSet documentation is egregiously ambiguous, but it seems like it should not be possible to hit this case, given the previous case?

Also, we probably need a corresponding change to the Deployment-tracking code?

Reply:
I played through an upgrade, and saw this case (generation > observedGeneration) after the operator (well, me at the CLI) had updated the daemonset and before the daemonset controller had reacted to it. 

And, yes, the deployment controller also needs some tweaks.

Reply:
> after the operator had updated the daemonset and before the daemonset controller had reacted to it.

right, but that's very transient; the daemonset controller should then react to it, at which point `ds.Status.UpdatedNumberScheduled < ds.Status.DesiredNumberScheduled` will be true